controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorGroupImpl.java

```diff
@@ -204,8 +199,41 @@ protected void shutDown() {
         synchronized (lock) {
             long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "shutDown");
             try {
-                cleanUpCheckpointStore();
+                // If any of the following operations error out, the fact is just logged.
+                // And some other controller process would clean-up the reader group data
+                // its readers and their position objects from checkpoint store.
+
+                // Seal the Reader Group
+                Map<String, Position> readerPositions = ImmutableMap.of();
+                try {
+                    readerPositions = sealReaderGroup(actorSystem.getProcess());
+                } catch (CheckpointStoreException e) {
+                    log.warn("Error sealing Reader Group {}.", readerGroup.getGroupName(), e);
+                }
+
+                // Initiate stop on all event processor cells
+                for (EventProcessorCell<T> cell : eventProcessorMap.values()) {
+                    log.info("Terminating event processor cell: {}", cell);
+                    cell.stopAsync();
+                }
+                // Await termination of cells
+                for (EventProcessorCell<T> cell : eventProcessorMap.values()) {
+                    try {
+                        cell.awaitTerminated();
+                        log.info("Termination of event processor cell: {} completed successfully.", cell);
+                    } catch (Exception e) {
+                        log.warn("Failed terminating event processor cell {}.", cell, e);
+                    }
+                }
+
+                // Finally, clean up Reader Group data from Checkpoint store.
+                try {
+                    cleanupReaderGroup(actorSystem.getProcess(), readerPositions);
+                } catch (CheckpointStoreException e) {
+                    log.warn("Error removing Reader Group {} from checkpoint store. " + readerGroup.getGroupName(), e);
```

Comment:
I think this log is wrong, as it is concatenating `readerGroup.getGroupName()` and also placing a `{}` for it. You may just replace `+` by `,`.

Reply:
Yes. Thanks for catching this. fixed.